pkg/storage/crd/storage_interface.go

```diff
@@ -0,0 +1,547 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crd
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/golang/glog"
+	scmeta "github.com/kubernetes-incubator/service-catalog/pkg/api/meta"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1alpha1"
+	"golang.org/x/net/context"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apiserver/pkg/storage"
+	"k8s.io/apiserver/pkg/storage/etcd"
+	"k8s.io/apiserver/pkg/storage/storagebackend/factory"
+	restclient "k8s.io/client-go/rest"
+)
+
+var (
+	errNotImplemented = errors.New("not implemented for third party resources")
+)
+
+type store struct {
+	// TODO (nilebox): review the usage of "hasNamespace" flag
+	hasNamespace     bool
```

Comment:
in the original TPR code, if this flag is set, you wipe the namespace from the objects even if they have a namespace. I haven't copied these code blocks yet.
@arschles why do we need this?

Reply:
because some service-catalog resources do not have namespaces (i.e. `Broker`), but all TPR instances need namespaces. This flag indicated if the TPRs created by the interface should go into the specified namespace of the default global namespace